packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts

```diff
@@ -53,13 +54,12 @@ export class AwsCliCompatible {
       () => new AWS.EnvironmentCredentials('AMAZON'),
     ];
 
-    if (await fs.pathExists(credentialsFileName())) {
-      // Force reading the `config` file if it exists by setting the appropriate
-      // environment variable.
-      await forceSdkToReadConfigIfPresent();
-      sources.push(() => profileCredentials(implicitProfile));
-      sources.push(() => new AWS.ProcessCredentials({ profile: implicitProfile }));
-    }
+    // Force reading the `config` file if it exists by setting the appropriate
+    // environment variable.
+    await forceSdkToReadConfigIfPresent();
+    sources.push(() => profileCredentials(implicitProfile));
```

Comment:
Seems like the list of sources is duplicated in two places in the code. Maybe find a way to reuse?